src/qibolab/_core/instruments/emulator/emulator.py

```diff
@@ -61,6 +61,8 @@ def play(
         options: ExecutionParameters,
         sweepers: list[ParallelSweepers],
     ) -> dict[int, Result]:
+        # convert align to delays
+        sequences = [seq.align_to_delays() for seq in sequences]
```

Comment:
```suggestion
        sequences = (seq.align_to_delays() for seq in sequences)
```

It will be iterated only once, so a generator is sufficient (a generator avoids iterating the list ahead of time, to construct another one that will be iterated once more - instead, the iteration will be lazy).